src/users/models/add_human_user_request.rs

```diff
@@ -0,0 +1,250 @@
+/*
+ * User Service
+ *
+ * This API is intended to manage users in a ZITADEL instance.
+ *
+ * OpenAPI spec version: 2.0
+ * Contact: hi@zitadel.com
+ * Generated by: https://github.com/swagger-api/swagger-codegen.git
+ */
+
+use serde::{Deserialize, Serialize};
+#[allow(unused_imports)]
+use serde_json::Value;
+
+use crate::users::models;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AddHumanUserRequest {
+	/// optionally set your own id unique for the user.
+	#[serde(rename = "userId")]
+	user_id: Option<String>,
+	/// optionally set a unique username, if none is provided the email will be
+	/// used.
+	#[serde(rename = "username")]
+	username: Option<String>,
+	#[serde(rename = "organization")]
+	organization: Option<models::Organization>,
+	#[serde(rename = "profile")]
+	profile: models::SetHumanProfile,
+	#[serde(rename = "email")]
+	email: models::SetHumanEmail,
+	#[serde(rename = "phone")]
+	phone: Option<models::SetHumanPhone>,
+	#[serde(rename = "metadata")]
+	metadata: Option<Vec<models::SetMetadataEntry>>,
+	#[serde(rename = "password")]
+	password: Option<models::Password>,
+	#[serde(rename = "hashedPassword")]
+	hashed_password: Option<models::HashedPassword>,
+	#[serde(rename = "idpLinks")]
+	idp_links: Option<Vec<models::IdpLink>>,
+	/// An Implementation of RFC 6238 is used, with HMAC-SHA-1 and time-step of
+	/// 30 seconds. Currently no other options are supported, and if anything
+	/// different is used the validation will fail.
+	#[serde(rename = "totpSecret")]
+	totp_secret: Option<String>,
+}
+
+impl AddHumanUserRequest {
+	pub fn new(
+		profile: models::SetHumanProfile,
+		email: models::SetHumanEmail,
+	) -> AddHumanUserRequest {
+		AddHumanUserRequest {
+			user_id: None,
+			username: None,
+			organization: None,
+			profile: profile,
+			email: email,
+			phone: None,
+			metadata: None,
+			password: None,
+			hashed_password: None,
+			idp_links: None,
+			totp_secret: None,
+		}
+	}
+
+	pub fn set_user_id(&mut self, user_id: String) {
+		self.user_id = Some(user_id);
+	}
+
+	pub fn with_user_id(mut self, user_id: String) -> AddHumanUserRequest {
```

Comment:
It's typical to offer *either* an owned *or* a referenced set of methods, and just name them the attribute they are setting. See e.g. [`reqwest::ClientBuilder`](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html).

One of the crates that provide macros for this would handle it more nicely anyway, though.